### PR TITLE
[SYCL][CUDA] Support device ID and UUID in the CUDA Plugin

### PR DESCRIPTION
**Pull request by zjin-lcf**
This PR tries to enable the accesses to device ID and UUID with the CUDA driver APIs.  Thank you for your review.

Reference
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_device_info.md

### CODE REVIEW

**Comment by zjin-lcf** [comment]
The code changes reported by lint are not consistent with the coding style in the plugin.

**Comment by abagusetty** [comment]
> The code changes reported by lint are not consistent with the coding style in the plugin.

I believe the CI uses `clang-format-v13` for formatting. Were you using anything other versions ?

**Comment by zjin-lcf** [comment]
No, I just looked at other codes in the plugin, and then follow that style.

**Comment by npmiller** [comment]
The linter is consistent, there's other parts of the plugins that are formatted like it is suggesting here, and all of the plugin code has gone through the linter.

It's just that the rules it uses to format are not always directly obvious, for example for this:

```diff
     sycl::detail::pi::assertion(
-        cuDeviceGetAttribute(&value,
-                             CU_DEVICE_ATTRIBUTE_PCI_DEVICE_ID,
+        cuDeviceGetAttribute(&value, CU_DEVICE_ATTRIBUTE_PCI_DEVICE_ID,
                              device->get()) == CUDA_SUCCESS);
```

If you look at other places that have it like this, they have enums with longer names that wouldn't fit on the first line in 80 columns.

Line breaking rules can get very tricky in C++, and that's why we should trust the linter even if it's not always the prettiest.

So could you update the code as the linter is suggesting? And in general I suggest using `clang-format`, you can even build it as part of DPC++ `ninja clang-format`.

Other than that the patch looks good to me :+1:

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 1]
```diff
@@ -1994,8 +1994,36 @@ pi_result cuda_piDeviceGetInfo(pi_device device, pi_device_info param_name,
...
+        cuDeviceGetAttribute(&value,
+                             CU_DEVICE_ATTRIBUTE_PCI_DEVICE_ID,
+                             device->get()) == CUDA_SUCCESS);
+    sycl::detail::pi::assertion(value >= 0);
+    return getInfo(param_value_size, param_value, param_value_size_ret, value);
+  }
+
+  case PI_DEVICE_INFO_UUID: {
+    int driver_version = 0;
+    cuDriverGetVersion(&driver_version);
```
Comment (abagusetty) [inline sycl/plugins/cuda/pi_cuda.cpp:2009]: @zjin-lcf @npmiller Do we need a API check for this. Also would you suggest to replace `sycl::detail::pi::assertion` to `PI_CHECK_ERROR` ?

**Review by npmiller — APPROVED** [review]

**Review by abagusetty — APPROVED** [review]
LGTM!

**Comment by bader** [comment]
@zjin-lcf, thanks for contributing this!

**Comment by zjin-lcf** [comment]
I want to thank the reviewers for their comments.

**Pull request merged**